File: algorithms/sort/bubbleSort.test.ts
import bubbleSort from "./bubbleSort.ts";

function testBubbleSort() {
  let nums = [5, -2, 12, 0, 9], expected = [-2, 0, 5, 9, 12];
  bubbleSort(nums);
  console.assert(
    JSON.stringify(nums) === JSON.stringify(expected),
    `expect bubbleSort(...) to be: ${expected}, got instead: ${nums}`,
  );

  nums = [1, -2, 10, -6], expected = [-6, -2, 1, 10];
  bubbleSort(nums);
  console.assert(
    JSON.stringify(nums) === JSON.stringify(expected),
    `expect bubbleSort(...) to be: ${expected}, got instead: ${nums}`,
  );
}

testBubbleSort();

File: algorithms/sort/bubble_sort_test.go
package sort

import (
	"reflect"
	"testing"
)

func TestBubbleSortInt(t *testing.T) {
	expected, nums := []int{-2, 0, 5, 9, 12}, []int{12, 0, 9, -2, 5}
	BubbleSortInt(nums, func(prev, next int) bool {
		return prev < next
	})
	if !reflect.DeepEqual(expected, nums) {
		t.Errorf("expect BubbleSort(...) to be [%v], got instead [%v]", expected, nums)
	}

	expected, nums = []int{10, 1, -2, -6}, []int{1, -2, 10, -6}
	BubbleSortInt(nums, func(prev, next int) bool {
		return prev > next
	})
	if !reflect.DeepEqual(expected, nums) {
		t.Errorf("expect BubbleSort(...) to be [%v], got instead [%v]", expected, nums)
	}
}
